### ISSUE TITLE
Problem AZERTY french keyboard layout

### ISSUE THREAD

**Issue by taldeguer** [Swordfish90/cool-retro-term#523]
There is a problem with the AZERTY french disposition.  the A key gives Q, but Q also gives Q. Same for M key, M = M , ?/, = M.
It seems both QWERTY/AZERTY are mixed. Is it possible to choose another layout and how ?

**Comment by taldeguer** [comment]
Also, I forgot to mention that I'm running OSX Mojave

**Comment by Jefwillems** [comment]
Same problem here

**Comment by dairaine** [comment]
+1

**Comment by mblarsen** [comment]
I'm experiencing this with Colemak as well.

**Comment by AlvaroAT95** [comment]
Same with Spanish QWERTY. Only happens in Mac OS using any version above v1.0.1. I've tried v1.1.1 in Linux and all is working properly.

**Comment by solsticedhiver** [comment]
Isn't the issue simply that QWERTY is the only layout whatever the one already set in your desktop ?

You could change your layout in your desktop, it will always be QWERTY in CRT

**Comment by Alvaro-AT** [comment]
Yes, I think that's the problem, but only happens in Mac OS.

**Comment by solsticedhiver** [comment]
No, no. If I am talking about desktop, it's because I am using linux. Ubuntu 19.04/wayland.

So that defintivly does not happen only in MacOS

**Comment by Jefwillems** [comment]
> Isn't the issue simply that QWERTY is the only layout whatever the one already set in your desktop ?
> 
> You could change your layout in your desktop, it will always be QWERTY in CRT

That's the problem, would you change your keyboard layout for every app you use? There should just be support for azerty and other layouts in this app, hence the issue.

**Comment by lraulin** [comment]
>There should just be support for azerty and other layouts in this app, hence the issue.

The app should use the OS-selected keyboard layout like every single other app, and like CRT apparently did until the most recent update.

Interesting that there's a report of it happening in Linux too... Like others, I can't use any non-QWERTY layout in OSX Mojave, but have never seen this problem on Linux.

**Comment by nicolas-b12** [comment]
Same here, OSX Mojave, AZERTY Keyboard

**Comment by slupartoo** [comment]
Same issue, with more informations.

My config is OSX Mojave and AZERTY Keyboard.

The first and third rows are QWERTY, they render `qwertyuiop` and `zxcvbm`. 
Whereas the second row renders `qsdfghjklm` which is the AZERTY configuration.

So I can't type the letter `a` 😞

**Comment by jbltx** [comment]
Same problem here on OSX Mojave, AZERTY French keyboard layout.

**Comment by khl3o** [comment]
same here, macOS Mojave ands AZERTY keyboard.

Also:
- The cmd + c / cmd + v doesn't work
- Touch a returns a 'q' and touch q return also a 'q'

I'm sad, it looks so nice...

**Comment by laubster** [comment]
Steps to demonstrate the problem (these work under the linux neon distro - may need tweaking for others):
1) leave desktop environment keyboard settings at the default
2) in terminal run "setxkbmap dvorak" (or maybe some azerty if it's installed)
3) start the cool-retro-term AppImage
4) notice that typing reflects qwerty instead of the requested alternate keyboard

**Comment by BuonOmo** [comment]
With bépo (French Dvorak) I end end having a qwerty Keyboard... (Mojave as well)

**Comment by diedummydie** [comment]
Same problem with Colemak on macOS 10.14 Mojave

**Comment by ChristopherJdL** [comment]
Same problem with macOS Catalina in 2020

**Comment by SirCthulhu** [comment]
Same Big Sur 2021 ! how to fix it ?
